flight/Modules/VtolPathFollower/vtol_follower_control.c

```diff
@@ -58,6 +58,131 @@ static VtolPathFollowerSettingsData guidanceSettings;
 static AltitudeHoldSettingsData altitudeHoldSettings;
 struct pid vtol_pids[VTOL_PID_NUM];
 
+// Constants used in deadband calculation
+static float vtol_path_m=0, vtol_path_r=0, vtol_end_m=0, vtol_end_r=0;
+
+static int32_t vtol_follower_control_simple(const float dT,
+	const float *hold_pos_ned, bool landing, bool update_status);
+
+/**
+ * Interpolate values (groundspeeds, altitudes) over flight legs
+ * @param[in] fraction how far we are through the leg
+ * @param[in] beginVal the configured value for the beginning of the leg
+ * @param[in] endVal the configured value for the end of the leg
+ * @returns the interpolated value
+ *
+ * Simple linear interpolation with clipping to ends (fraction>=0, <=1).
+ */
+static float vtol_interpolate(const float fraction, const float beginVal,
+	const float endVal) {
+	return beginVal + (endVal - beginVal) * bound_min_max(fraction, 0, 1);
+}
+
+/**
+ * Calculate pythagorean magnitude of a vector.
+ * @param[in] v pointer to a float array
+ * @param[in] n length of the amount to take the magnitude of
+ * @returns Root of sum of squares of the vector
+ *
+ * Note that sometimes we only take the magnitude of the first 2 elements
+ * of a 3-vector to get the horizontal component.
+ */
+static float vtol_magnitude(const float *v, int n) {
```

Comment:
this should probably be moved to a math library and just called vector magnitude. e.g. https://github.com/TauLabs/TauLabs/blob/next/flight/Libraries/math/coordinate_conversions.h#L70